PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidServerAdapter.java

```diff
@@ -560,14 +573,38 @@ private JSONArray getImp() throws NoContextException {
                     }
                     banner.put("format", format);
                     imp.put("banner", banner);
-                } else {
+                } else if (adType.equals(AdType.BANNER)) {
                     JSONObject banner = new JSONObject();
                     JSONArray format = new JSONArray();
                     for (AdSize size : requestParams.getAdSizes()) {
                         format.put(new JSONObject().put("w", size.getWidth()).put("h", size.getHeight()));
                     }
                     banner.put("format", format);
                     imp.put("banner", banner);
+                } else if (adType.equals(AdType.VIDEO) || adType.equals(AdType.VIDEO_INTERSTITIAL)) {
+
+                    JSONObject video = new JSONObject();
+                    video.put("mimes", new JSONArray().put("video/mp4"));
+                    video.put("protocols", new JSONArray().put(7));
```

Comment:
Should this be configurable by the publisher? I also read the doc that there should be more fields.

Reply:
Typically a publisher sets these to what is supported by the video player. Since we are providing the video player support, we are defining these values for them. If they wish to override them, in the subsequent release, we can read them from Prebid Server using this Intent to implement where a publisher can set them using Stored Requests.:
 https://github.com/prebid/prebid-server/issues/1075

The only real setting I imagine a publisher changing is playbackmethod and possibly protocols. This first pass we are hardcoding the values with the goal of keeping all these attributes in Stored Request for more flexibility vs exposing these values in SDK. Changing values would be extremely difficult in SDK if we expose them.

Reply:
To provide additional context, the device player does not support all mime types per: https://developer.android.com/guide/topics/media/media-formats

We don't want publisher misrepresenting the file types for the player. We selected mp4 since this most demand mime type. Allowing a publisher to set this can cause ads not to render.